configurator_wizard_in_product/wizard/product_configurator.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, api, _
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from odoo.exceptions import ValidationError
+from datetime import datetime
+
+
+class ProductConfigurator(models.TransientModel):
+    _inherit = 'product.configurator'
+
+    @api.multi
+    def get_state_selection(self):
+        """ copied from product_configurator_wizard.product_configurator.py
+            to add condition  to modify states if wizard is called
+            from product.template object """
+
+        steps = [('select', "Select Template")]
+
+        # Get the wizard id from context set via action_next_step method
+        wizard_id = self.env.context.get('wizard_id')
+
+        if not wizard_id:
+            return steps
+
+        wiz = self.browse(wizard_id)
+
+        open_lines = wiz.product_tmpl_id.get_open_step_lines(
+            wiz.value_ids.ids)
+
+        if open_lines:
+            open_steps = open_lines.mapped(
+                lambda x: (str(x.id), x.config_step_id.name)
+            )
+            if self.env.context.get('active_model') == 'product.template':
+                steps = open_steps if (wiz.product_id or wiz.product_tmpl_id) else steps + open_steps
+            else:
+                steps = open_steps if wiz.product_id else steps + open_steps
+        else:
+            if self.env.context.get('active_model') == 'product.template':
+                steps = [('configure', 'Configure')]
+            else:
+                steps.append(('configure', 'Configure'))
+        return steps
+
+    @api.multi
+    def action_config_done(self):
+        """This function is copied from product_configurator_wizard.ProductConfigurator
+           added this function to check active_model=='product.template' to ignore running not necessary code"""
+        custom_vals = {
+            l.attribute_id.id:
+                l.value or l.attachment_ids for l in self.custom_value_ids
+        }
+
+        # This try except is too generic.
+        # The create_variant routine could effectively fail for
+        # a large number of reasons, including bad programming.
+        # It should be refactored.
+        # In the meantime, at least make sure that a validation
+        # error legitimately raised in a nested routine
+        # is passed through.
+        domain = [['product_tmpl_id', '=', self.product_tmpl_id.id]]
+
+        template_products = self.env['product.product'].search(domain)
+        product_found = False
+
+        if self.product_tmpl_id.reuse_variant:
+            for product in template_products:
+                if product.attribute_value_ids == self.value_ids:
+                    product_found = product
+                    break
+
+        if product_found:
```

Comment:
Bad bad bad!

we are modifying the current variant, not creating / finding another